active/Breadcrumb/Breadcrumb.md

```diff
@@ -0,0 +1,225 @@
+
+XAML
+```XAML
+<Breadcrumb x:Name="Breadcrumb1" /> 
+```
+
+C#
+```cs
+Breadcrumb1.ItemsSource = new string[]{ "Home", "Documents", "Design", "Northwind", "Images"}
+```
+
+If the last item in a Breadcrumb is text and is too long to fit it will be clipped
+(if the item type is a string it will be trimmed, meaning shortened and replaced with an ellipsis).
+
+![Breadcrumb_crumbled with last node truncated](images/Breadcrumb_truncation.PNG)
+
+#### Responding to clicks
+
+This example shows an app with a Breadcrumb across the top, and a content area on the rest of the Window.
+The Breadcrumb shows a list of items. Clicking on an item navigates the content back, and updates
+the Breadcrumb.
+
+```xaml
+<Grid RowDefinitions="Auto,*">
+    <Breadcrumb ItemClicked="CrumbClicked" ItemsSource="{x:Bind Items, Mode=OneWay}" />
+    
+    <ContentControl Grid.Row="1" Content="{x:Bind CurrentItem, Mode=OneWay}">
+        ...
+    </ContentControl>
+</Grid>
+```
+
+```csharp
+public sealed partial class MainPage : Page
+{
+    public ObservableCollection<Items> Items { get; }
+
+    public ItemClickEventArgs Current Item { get; set;}
+
+    void CrumbClicked(Breadcrumb sender, BreadcrumbItemClickedEventArgs e)
+    {
+        for(int i = Items.Count-1; i > e.Index; i++)
+        {
+            Items.RemoveAt(i);
+        }
+
+        CurrentItem = Items[i - 1];
+    }
+
+    void NavigateToItem(Item item)
+    {
+        Items.Add(item);
+        CurrentItem = item;
+    }
+}
+```
+
+#### Breadcrumb.ItemTemplate property
+
+Gets or sets the template to display an item
+
+The Breadcrumb ItemTemplate is used to override the default display of the
+individual items in the Breadcrumb control.
+
+The following uses the ItemTemplate so that the Breadcrumb items show the Name property of the items:
+
+```xaml
+<Breadcrumb ItemsSource='{x:Bind Items}'>
+    <Breadcrumb.ItemTemplate>
+        <DataTemplate x:DataType='BreadcrumbItem`>
+            <TextBlock Text='{x:Bind Name}/>
+        </DataTemplate>
+    </Breadcrumb.ItemTemplate>
+</Breadcrumb>
+```
+
+Notes:
+* You can use a BreadcrumbItem as the root of the DataTemplate
+* If no ItemTemplate is supplied, the Breadcrumb control will convert to a string.
+
+## Other Breadcrumb members
+
+| Name | Description | Default |
+| :---------- | :------- | :------- |
+| ItemsSource | Gets or sets the content of the Breadcrumb |
+| ItemClicked | Raised when a user interaction causes a jump to the CurrentItem.  |
+
+_Spec note: the V2 of Breadcrumb will add flyouts from chevrons to view children of a node.
+Additional properties will be added to the API to enable this functionality._
+
+# API Details
+
+**Windows.UI.Xaml.Controls namespace**
+
+```cs
+[webhosthidden]
+runtimeclass BreadcrumbItemClickedEventArgs
+{
+    Int32 Index { get; };
+    Object Item { get; };
+}
+
+[webhosthidden]
+unsealed runtimeclass BreadcrumbItem : Windows.UI.Xaml.Controls.ContentControl
+{
+    BreadcrumbItem();
+}
```

Comment:
Why do we need the BreadcrumbItem class at all? The ItemsSource is a collection of Objects, which can be anything. If the app doesn't have anything convenient, it can use a String (or if it really wants an element, a TextBlock).

I'm guessing this is for parallel construction with ListBoxItem, ComboBoxItem, etc. But ListBox, ComboBox, etc. are ItemsControls and have an Items collection, so consumers which do not use data binding have something to stuff into the Items collection.

But Breadcrumb is not an ItemsControl. It doesn't have an Items collection. You *must* data-bind. So there's no need for the "lazy" item class.

(Should Breadcrumb be an ItemsControl?)

(Having a BreadcrumbItem control makes the class name BreadcrumbItemCilckedEventArgs ambiguous. Is it Breadcrumb+ItemClicked+EventArgs? Or is it BreadcrumbItem+Clicked+EventArgs?)

Reply:
BreadcrumbItem is what provides the default visuals for pointer pressed/over etc, so you need some container that can provide those visuals. That is similar to ListViewItem etc as you mentioned. You can manually add items to the collection that are of type BreadcrumbItem and we will use that as opposed to us automatically creating them if you provide data objects.

BreadCrumbItem+Clicked+EventArgs . I agree, that its not obvious which one we are talking about from the name.